core/src/fnc/array.rs

```diff
@@ -328,14 +391,27 @@ pub fn remove((mut array, mut index): (Array, i64)) -> Result<Value, Error> {
 	Ok(array.into())
 }
 
+pub fn repeat((value, count): (Value, i64)) -> Result<Value, Error> {
+	if count < 0 {
+		return Err(Error::InvalidArguments {
+			name: String::from("array::repeat"),
+			message: format!(
+				"Argument 2 was the wrong type. Expected a positive number but found {count}"
+			),
+		});
+	}
+
+	Ok(Array(std::iter::repeat(value).take(count as usize).collect()).into())
```

Comment:
To reduce the chance of memory exhaustion if `count` is very large, consider limiting the length of the resulting array like `string::repeat`: https://github.com/surrealdb/surrealdb/blob/1baf7848db4894b67f80c787b4f863eedbae3452/core/src/fnc/string.rs#L63-L66

Reply:
Done.